src/openstudio_lib/ScheduleOthersController.cpp

```diff
@@ -0,0 +1,109 @@
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "ScheduleOthersController.hpp"
+#include "ScheduleOthersView.hpp"
+
+#include <openstudio/model/ScheduleConstant.hpp>
+#include <openstudio/model/ScheduleConstant_Impl.hpp>
+#include <openstudio/model/ExternalFile.hpp>
+#include <openstudio/model/ExternalFile_Impl.hpp>
+#include <openstudio/model/ScheduleFile.hpp>
+
+#include <openstudio/utilities/core/Logger.hpp>
+
+#include <openstudio/utilities/idd/IddEnums.hxx>
+
+#include <QMessageBox>
+#include <QFileDialog>
+#include <QDir>
+
+namespace openstudio {
+
+ScheduleOthersController::ScheduleOthersController(const model::Model& model) : ModelSubTabController(new ScheduleOthersView(model), model) {}
+
+void ScheduleOthersController::onAddObject(const openstudio::IddObjectType& iddObjectType) {
+  switch (iddObjectType.value()) {
+    case IddObjectType::OS_Schedule_Constant: {
+      openstudio::model::ScheduleConstant(this->model());
+      break;
+    }
+    case IddObjectType::OS_Schedule_Compact: {
+      QMessageBox message(subTabView());
+      message.setText("Creation of Schedule:Compact is not supported, you should use a ScheduleRuleset instead");
+      message.exec();
+      return;
+    }
+    case IddObjectType::OS_Schedule_File: {
+      QString selfilter = tr("CSV Files(*.csv)");
+      QString fileName = QFileDialog::getOpenFileName(this->subTabView(), tr("Select External File"), QDir::homePath(),
+                                                      tr("All files (*.*);;CSV Files(*.csv);;TSV Files(*.tsv)"), &selfilter);
+      if (!fileName.isNull()) {
+        boost::optional<model::ExternalFile> e_ = model::ExternalFile::getExternalFile(this->model(), fileName.toStdString());
+        if (e_) {
+          model::ScheduleFile(*e_, 1, 1);
+        }
+        break;
+      }
```

Comment:
@jmarrec can you move the break down one level, right now if you cancel out of the file dialog the switch statement falls through into the default section and calls `LOG_FREE_AND_THROW`

Reply:
good catch thanks